wrap.sh

```diff
@@ -178,26 +179,42 @@ while getopts "r:w:e:abcdhmnpuv" opt; do
     env_vars+=(DBUS_SESSION_BUS_ADDRESS)
     ;;
 
-  # grant desktop access, wayland, X11, DRI
+  # grant desktop access (Wayland or X11) and rendering hardware access
   d)
-    bwrap_opts+=(--bind "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY" "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY")
+    if [ -n "${WAYLAND_DISPLAY:-}" ] && [ -n "${XDG_RUNTIME_DIR:-}" ]; then
+      # Using Wayland: bind the Wayland display socket
+      bwrap_opts+=(--bind "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY" "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY")
+    elif [ -n "${DISPLAY:-}" ]; then
+      # Using X11: bind the X11 socket directory
+      # The standard location is usually /tmp/.X11-unix.
+      if [ -d "/tmp/.X11-unix" ]; then
+        bwrap_opts+=(--ro-bind "/tmp/.X11-unix" "/tmp/.X11-unix")
+      fi
+
+      # Bind the .Xauthority file so that the authorization data is available.
+      if [ -n "XAUTHORITY" ]; then
```

Comment:
```suggestion
      if [ -n "${XAUTHORITY}" ]; then
```

I think the idea was to check if the env var is non empty?

Reply:
Actually, I am a bit confused by this code. Are we checking for an env var or for the ~/.Xauthority file?

Reply:
my bash is embarrassingly bad. Pretty sure we want the interpolated one (good catch by you). Otherwise it is literally checking if the string "XAUTHORITY" is non-zero in length, which is trivially true.